app/src/main/java/org/dosomething/letsdothis/tasks/CampaignSignUpTask.java

```diff
@@ -0,0 +1,54 @@
+package org.dosomething.letsdothis.tasks;
+import android.content.Context;
+
+import org.dosomething.letsdothis.data.CampaignActions;
+import org.dosomething.letsdothis.network.NetworkHelper;
+import org.dosomething.letsdothis.network.models.RequestCampaignSignup;
+import org.dosomething.letsdothis.network.models.ResponseCampaignSignUp;
+import org.dosomething.letsdothis.utils.AppPrefs;
+
+import co.touchlab.android.threading.tasks.Task;
+
+/**
+ * Created by izzyoji :) on 6/26/15.
+ */
+//todo should this be persisted
+public class CampaignSignUpTask extends Task
+{
+    private int campaignId;
+
+    public CampaignSignUpTask(int campaignId)
+    {
+        this.campaignId = campaignId;
+    }
+
+    @Override
+    protected void onComplete(Context context)
+    {
+        super.onComplete(context);
+    }
+
+    @Override
+    protected void run(Context context) throws Throwable
+    {
+        if(CampaignActions.queryForId(context, campaignId) == null)
+        {
+            String sessionToken = AppPrefs.getInstance(context).getSessionToken();
+            RequestCampaignSignup requestCampaignSignup = new RequestCampaignSignup();
+            ResponseCampaignSignUp response = NetworkHelper.getNorthstarAPIService()
```

Comment:
doesnt work and crashes. add fixme if the api is not done and comment out the breaking call.